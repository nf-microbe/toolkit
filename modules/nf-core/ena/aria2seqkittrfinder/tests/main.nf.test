nextflow_process {

    name "Test Process ENA_ARIA2SEQKITTRFINDER"
    script "../main.nf"
    process "ENA_ARIA2SEQKITTRFINDER"

    tag "modules"
    tag "modules_nfcore"
    tag "ena"
    tag "ena/aria2seqkittrfinder"

    test("SRR11489779 + SRR11489784") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    [
                        "https://ftp.sra.ebi.ac.uk/vol1/analysis/ERZ114/ERZ11466704/SRR11489779.fasta.gz",
                        "https://ftp.sra.ebi.ac.uk/vol1/analysis/ERZ114/ERZ11466716/SRR11489784.fasta.gz"
                    ]
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    [
                        "https://ftp.sra.ebi.ac.uk/vol1/analysis/ERZ114/ERZ11466704/SRR11489779.fasta.gz",
                        "https://ftp.sra.ebi.ac.uk/vol1/analysis/ERZ114/ERZ11466716/SRR11489784.fasta.gz"
                    ]
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
