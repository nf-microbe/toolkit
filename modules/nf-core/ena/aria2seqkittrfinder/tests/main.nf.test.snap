{
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.trfinder.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.trfinder.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,fcc251bebbb031253b43d97b358c534f"
                ],
                "fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "test.trfinder.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test.trfinder.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,fcc251bebbb031253b43d97b358c534f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-11T08:37:34.654469937"
    },
    "SRR11489779 + SRR11489784": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.trfinder.fasta.gz:md5,dbafb1295aa3c6cbdd0b8742f4091933"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.trfinder.tsv:md5,e433c00d8f62031e3e3cd72ceabbcd9f"
                    ]
                ],
                "2": [
                    "versions.yml:md5,fcc251bebbb031253b43d97b358c534f"
                ],
                "fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "test.trfinder.fasta.gz:md5,dbafb1295aa3c6cbdd0b8742f4091933"
                    ]
                ],
                "tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test.trfinder.tsv:md5,e433c00d8f62031e3e3cd72ceabbcd9f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,fcc251bebbb031253b43d97b358c534f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-11T15:35:06.817897365"
    }
}